### PR TITLE
Changes needed to meet security requirements for LFG

### PR DESCRIPTION
**Pull request by fritzdj**
**Security related changes**
Supporting the ability to use KMS keys to encrypt S3 bucket contents
Supporting the ability to use KMS keys to encrypt RDS instance data
Supporting the ability to enable S3 and ELB/ALB access logging
Adding an optional DDoS / WAF broker to configure a web ACL for an app + enable Shield

**Other changes to support LFG apps**
Supporting Postgres extentions using the RDS cred broker
Adding a mysql56 profile to support 6.5 version instances

### CODE REVIEW

File: src/main/java/com/libertymutualgroup/herman/aws/ecs/EcsDefinitionParser.java [hunk 1]
```diff
@@ -123,6 +124,9 @@ private void patchJsonLogDefintionParseError(List<ContainerDefinition> container
 
         @JsonIgnore
         void setName(UlimitName name);
+
+        @JsonProperty("name")
+        void setName(String name);
```
Comment (fritzdj) [inline src/main/java/com/libertymutualgroup/herman/aws/ecs/EcsDefinitionParser.java]: @CWDoherty's change here that snuck in

File: src/main/java/com/libertymutualgroup/herman/aws/ecs/EcsPush.java [hunk 2]
```diff
@@ -218,7 +219,7 @@ public EcsPush(EcsPushContext context) {
 
         this.lambdaClient = AWSLambdaClientBuilder.standard()
             .withCredentials(new AWSStaticCredentialsProvider(context.getSessionCredentials()))
-            .withClientConfiguration(context.getAwsClientConfig())
+            .withClientConfiguration(new ClientConfiguration().withClientExecutionTimeout(300000).withSocketTimeout(300000))
```
Comment (fritzdj) [inline src/main/java/com/libertymutualgroup/herman/aws/ecs/EcsPush.java]: Maxing execution and socket timeout values to ensure broker lambdas don't run multiple times

File: src/main/java/com/libertymutualgroup/herman/aws/ecs/EcsPushDefinition.java [hunk 3]
```diff
@@ -62,6 +63,7 @@
     private String kmsKeyName;
     private String iamRole;
     private String albTimeout;
+    private List<Ulimit> ulimits;
```
Comment (fritzdj) [inline src/main/java/com/libertymutualgroup/herman/aws/ecs/EcsPushDefinition.java]: Also @CWDoherty's change

File: src/main/java/com/libertymutualgroup/herman/aws/ecs/broker/kms/KmsBroker.java [hunk 4]
```diff
@@ -126,6 +129,14 @@ public String brokerKey(AWSKMS client, KmsAppDefinition definition, List<Tag> ta
                 String.format("Error updating key policy for key ID %s. Policy: %s", keyId, policy), ex);
         }
 
+        GetKeyRotationStatusResult statusResult = client.getKeyRotationStatus(new GetKeyRotationStatusRequest().withKeyId(keyId));
```
Comment (c1phr) [inline src/main/java/com/libertymutualgroup/herman/aws/ecs/broker/kms/KmsBroker.java]: I know it generally seems like no one would want to do this, but you might want to support disabling key rotation as well if someone really doesn't want it.

File: src/main/java/com/libertymutualgroup/herman/aws/ecs/broker/s3/S3Broker.java [hunk 5]
```diff
@@ -69,24 +90,39 @@ public S3Broker(S3CreateContext context) {
...
-        String policy = null;
 
+        // Setup up policy and tags for the bucket
+        String policy = null;
         if (configuration.getPolicyName() != null) {
             FileUtil fileUtil = new FileUtil(context.getRootPath(), buildLogger);
             policy = fileUtil.findFile(configuration.getPolicyName(), false);
         }
+        TagSet tags = new TagSet();
+        tagMap.entrySet().stream().forEach(it ->
```
Comment (c1phr) [inline src/main/java/com/libertymutualgroup/herman/aws/ecs/broker/s3/S3Broker.java]: We do operations with tags pretty frequently, and TagSet isn't the greatest type to work with compared to HashMap. Definitely doesn't need to happen as part of this PR, but an idea for a future refactor: we may want to add some utility methods to do this Map -> TagSet conversion so we can work with Maps and then convert at the AWS touchpoint.

**Comment by fritzdj** [comment]
I will make the changes for optionally disabling key rotation + working with HashMap for keys in a following PR

**Pull request merged**